sqlmesh/core/snapshot/definition.py

```diff
@@ -127,6 +127,23 @@ class SnapshotIntervals(PydanticModel, frozen=True):
     def snapshot_id(self) -> SnapshotId:
         return SnapshotId(name=self.name, identifier=self.identifier)
 
+    @classmethod
+    def from_snapshot(cls, snapshot: Snapshot, **kwargs: t.Any) -> SnapshotIntervals:
+        return cls(
```

Comment:
Why not use `snapshot.copy(update=kwargs)`? https://docs.pydantic.dev/latest/usage/exporting_models/#modelcopy